dashboard/models.py

```diff
@@ -159,18 +153,31 @@ def get_supported_courses(self):
 
 
 class Course(models.Model):
-    id = models.BigIntegerField(primary_key=True, verbose_name="Unizin Course Id", db_column='id', editable=False)
+    id = models.BigIntegerField(primary_key=True, verbose_name="Course Id", db_column='id')
     canvas_id = models.BigIntegerField(verbose_name="Canvas Course Id", db_column='canvas_id')
-    term_id = models.ForeignKey(AcademicTerms, verbose_name="Term Id", on_delete=models.SET_NULL, db_column='term_id', null=True, db_constraint=False)
+    term = models.ForeignKey(AcademicTerms, verbose_name="Term", on_delete=models.SET_NULL, db_column="term_id", null=True, db_constraint=False)
     name = models.CharField(max_length=255, verbose_name="Name")
+    date_start = models.DateTimeField(verbose_name="Start Date and Time", null=True, blank=True)
+    date_end = models.DateTimeField(verbose_name="End Date and Time", null=True, blank=True)
 
     objects = CourseManager()
 
     def __str__(self):
         return self.name
 
+    def get_course_date_range(self):
+        if self.date_start is not None:
+            start = self.date_start
+        else:
+            start = self.term.date_start
+        if self.date_end is not None:
+            end = self.date_end
+        else:
+            end = self.term.get_correct_date_end()
+        return start, end
```

Comment:
A namedtuple would make this easier to read and work with. It's an 'anti-pattern' to use an ambiguous index.

https://docs.quantifiedcode.com/python-anti-patterns/readability/not_using_named_tuples_when_returning_more_than_one_value.html

Reply:
That makes sense to me. See latest commit.